metricbeat/module/elasticsearch/index_recovery/data_xpack.go

```diff
@@ -0,0 +1,87 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package index_recovery
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/elastic/beats/metricbeat/helper/elastic"
+	"github.com/elastic/beats/metricbeat/module/elasticsearch"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, content []byte) error {
+	var data map[string]interface{}
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		return err
+	}
+
+	var results []map[string]interface{}
+	for indexName, value := range data {
+		indexData, ok := value.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		value, ok = indexData["shards"]
```

Comment:
What is the reason here that `shards` etc. could be missing? 

Also worried about reuse of `value` here.

Reply:
> What is the reason here that shards etc. could be missing?

It should never happen. I was just being safe with the `ok` check.

Reply:
+1 on the `ok` check. Was mainly curious if in "normal" behaviour of the endpoint we expect this to happen. They way I understand your answer it's not the case.